jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java

```diff
@@ -689,7 +689,7 @@ private void parseBodyAndUpgrade() throws IOException
                         }
                     }
 
-                    proxyEndPoint = new ProxyEndPoint(endPoint, local, remote, tlvs, client == 0 ? null : EndPoint.SslSessionData.from(null, null, sslCipher, null));
+                    proxyEndPoint = new ProxyEndPoint(endPoint, local, remote, tlvs, (client & PP2_CLIENT_SSL) == 0 ? null : EndPoint.SslSessionData.from(null, null, sslCipher, null));
```

Comment:
I think this should be reverted, because `client == 2` or `client == 4` also mean that the connection was secured.